Userland/Libraries/LibCore/System.h

```diff
@@ -14,23 +14,31 @@
 #include <AK/StringView.h>
 #include <AK/Vector.h>
 #include <fcntl.h>
-#include <netdb.h>
-#include <poll.h>
-#include <pwd.h>
 #include <signal.h>
-#include <spawn.h>
-#include <sys/ioctl.h>
-#include <sys/resource.h>
-#include <sys/socket.h>
 #include <sys/stat.h>
-#include <sys/time.h>
-#include <sys/utsname.h>
-#include <sys/wait.h>
-#include <termios.h>
 #include <time.h>
-#include <utime.h>
 
-#if !defined(AK_OS_BSD_GENERIC) && !defined(AK_OS_ANDROID)
+#if !defined(AK_OS_WINDOWS)
+#    include <netdb.h>
+#    include <poll.h>
+#    include <pwd.h>
+#    include <spawn.h>
+#    include <sys/ioctl.h>
+#    include <sys/resource.h>
+#    include <sys/socket.h>
+#    include <sys/time.h>
+#    include <sys/utsname.h>
+#    include <sys/wait.h>
+#    include <termios.h>
+#    include <utime.h>
+#else
+#    define O_CLOEXEC O_NOINHERIT
+using sighandler_t = void (*)(int);
+using socklen_t = int;
+__declspec(dllimport) void __stdcall freeaddrinfo(struct addrinfo*);
```

Comment:
No real notes on the implementation here in System. Glad we could avoid windows.h.

I would rather the changes to System.h and adding SystemWindows.cpp be in a separate commit on this PR. Will make it easier to track adding features to SystemWindows.cpp as needed.

There's likely other calls in System.h that we don't actually need in ladybird, as this file was inherited from SerenityOS. We can remove those as we identify that there's no callers.

Reply:
I am assuming future changes to System.h/SystemWindows.cpp should be in a separate commit too.

Recent Core::System cleanup: 999c591e8340a62b96a48d7cb8a645cf596c0a79

Sadly I just noticed that .cpp which includes LibCore/System.h but doesn't include winsock2.h fails to compile because of struct addrinfo being incomplete type.
(winsock2.h includes windows.h)
I bypass this problem for now by excluding getaddrinfo declaration from compilation on Windows.

Reply:
If it makes sense in the end to move getaddrinfo/freeaddrinfo to its own header, I'm ok with that refactor. Core::System has always felt too large.